sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CsvOutputWriter.scala

```diff
@@ -47,3 +50,15 @@ class CsvOutputWriter(
 
   override def close(): Unit = gen.close()
 }
+
+object CsvOutputWriter {
+
+  def createUnivocityGenerator(
+      maxLength: Int,
+      writer: OutputStreamWriter,
+      dataSchema: StructType,
+      params: CSVOptions): UnivocityGenerator = this.synchronized {
+    NormalizedString.getCache.setMaxStringLength(maxLength)
```

Comment:
Looks like it will have to be a static SQL config .. if we have to expose this as an option.

Reply:
eh..may I ask why? is it a convention that we should expose all static configuration as static SQL config?

Reply:
Because of https://github.com/apache/spark/pull/30972#discussion_r550017618 .. this property looks it works as a global variable...